pkg/sessionctx/variable/session.go

```diff
@@ -1618,6 +1618,38 @@ type SessionVars struct {
 
 	// GroupConcatMaxLen represents the maximum length of the result of GROUP_CONCAT.
 	GroupConcatMaxLen uint64
+
+	// TablesBufferPool is a memory pool for table related memory allocation that aims to reuse memory
+	// and saves allocation
+	TablesBufferPool TablesBufferPool
```

Comment:
Better to comment about the lifetime of the reused-buffer or how to use it, like staging, statement or the whole session.

Since it's not related to a session variable, could we put it somewhere else considering its lifetime?

Reply:
By convention such data structures are directly put in `SessionVars`, like
https://github.com/pingcap/tidb/blob/ec6f56e82609659edb30dc309d2eb7615eb0f7b8/pkg/sessionctx/variable/session.go#L1324
https://github.com/pingcap/tidb/blob/ec6f56e82609659edb30dc309d2eb7615eb0f7b8/pkg/sessionctx/variable/session.go#L1198
https://github.com/pingcap/tidb/blob/ec6f56e82609659edb30dc309d2eb7615eb0f7b8/pkg/sessionctx/variable/session.go#L1490

Reply:
The `SessionVars` seems very bloated, which might not be a good design. We can leave the new fields here for now. Refactoring this part might be a matter for another PR.

Reply:
+1. That should be part of the session refactoring work.